tests/page_test.go

```diff
@@ -1699,3 +1699,59 @@ func TestShadowDOMAndDocumentFragment(t *testing.T) {
 		})
 	}
 }
+
+func TestPageTargetBlank(t *testing.T) {
+	t.Parallel()
+
+	tb := newTestBrowser(t, withHTTPServer())
+	tb.withHandler("/home", func(w http.ResponseWriter, _ *http.Request) {
+		_, err := w.Write([]byte(
+			`<!DOCTYPE html><html><head></head><body>
+				<h1>click please</h1><a href="/link" target="_blank">click me</a>
+			</body></html>`,
+		))
+		require.NoError(t, err)
+	})
+	tb.withHandler("/link", func(w http.ResponseWriter, _ *http.Request) {
+		_, err := w.Write(
+			[]byte(`<!DOCTYPE html><html><head></head><body><h1>you clicked!</h1></body></html>`),
+		)
+		require.NoError(t, err)
+	})
+
+	p := tb.NewPage(nil)
+
+	// Navigate to the page with a link that opens a new page.
+	opts := &common.FrameGotoOptions{
+		Timeout: common.DefaultTimeout,
+	}
+	resp, err := p.Goto(tb.url("/home"), opts)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	// Current page count should be 1.
+	pp := p.Context().Pages()
+	assert.Equal(t, 1, len(pp))
+
+	// This link should open the link on a new page.
+	err = p.Click("a[href='/link']", common.NewFrameClickOptions(p.Timeout()))
+	require.NoError(t, err)
+
+	// Page count should now be 2.
+	for i := 0; i < 5; i++ {
+		pp = p.Context().Pages()
+		if len(pp) == 2 {
+			break
+		}
+		time.Sleep(time.Millisecond * 500)
+	}
+	assert.Equal(t, 2, len(pp))
```

Comment:
Shouldn't this be `require`. Otherwise, the code below might panic.

Reply:
Yeah, good point. So I always saw `assert` as a way to inform readers that this is something that is key to the test, whereas `require` is something that isn't really being tested in this test but should pass still.

That's why I used `assert` here.

Happy to change this to `require` if you have a strong opinion.

Reply:
No strong opinion. It's OK.

Reply:
It's been changed now so I think the `assert` should still be ok since we're no longer working with `pp` beyond checking it's length. I'll keep this in mind though `require` vs `assert` 👍